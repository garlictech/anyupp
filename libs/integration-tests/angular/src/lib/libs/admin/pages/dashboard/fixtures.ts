import { ProductComponentSetType, ProductType } from '@bgap/domain';

export const printableOrder = {
  id: 'seeded_order_id_6',
  createdAt: '2022-01-31T15:04:37.521Z',
  userId: 'seeded_user_id_6',
  unitId: 'seeded_unit_c1_g1_1_id',
  orderNum: '000001',
  items: [
    {
      productId: 'seeded_chain_product_c1_1_id',
      variantId: 'seeded_chain_product_variant_c1_p1_1_id',
      created: null,
      productName: {
        en: 'Hamburger',
        de: 'Hamburger',
        hu: 'Hamburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['egg', 'gluten', 'mustard', 'milk', 'soya', 'fish', 'sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_2_id',
      variantId: '16853de0-528d-11ec-90c0-fdfe82b3802d',
      created: null,
      productName: {
        en: 'Fishburger',
        de: 'Fishburger',
        hu: 'Halburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: [
        'lupin',
        'molluscs',
        'crustaceans',
        'egg',
        'soya',
        'sesame',
        'fish',
        'gluten',
      ],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_3_id',
      variantId: 'd22dee40-528b-11ec-afe3-15e1afd659e2',
      created: null,
      productName: {
        en: 'Sajtburger',
        de: 'Sajtburger',
        hu: 'Sajtburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
  ],
  paymentMode: {
    type: 'cash',
    caption: null,
    method: 'cash',
  },
  statusLog: [
    {
      userId: 'test-alice',
      status: 'placed',
      ts: 1627909024677,
    },
  ],
  archived: false,
  takeAway: false,
  place: {
    seat: '02',
    table: '01',
  },
  paymentIntention: null,
  transactionStatus: 'success',
  transactionId: 'seeded_transaction_id_6',
  unpayCategory: null,
  orderMode: 'instant',
  servingMode: 'inPlace',
  serviceFee: {
    currency: 'HUF',
    netPrice: 200,
    taxPercentage: 27,
  },
  packagingSum: null,
  sumPriceShown: {
    currency: 'HUF',
    pricePerUnit: 298.2,
    priceSum: 2982,
    tax: 27,
    taxSum: 633.96,
  },
  rating: {
    key: 'RATING KEY',
    value: 3,
  },
  hasRated: true,
  tip: null,
  tipTransactionStatus: null,
  tipTransactionId: null,
  updatedAt: '2022-01-31T15:04:37.521Z',
  transaction: {
    id: 'seeded_transaction_id_6',
    userId: 'test-alice',
    orderId: 'seeded_order_id_6',
    type: 'cash',
    total: 0,
    currency: 'HUF',
    status: 'waiting_for_payment',
    externalTransactionId: null,
    invoiceId: null,
    receiptId: null,
    createdAt: '2022-01-31T15:04:37.267Z',
    updatedAt: '2022-01-31T15:04:37.267Z',
    paymentMethodId: null,
    user: {
      id: 'test-alice',
      name: 'Mekk Elek',
      email: 'testuser+alice@anyupp.com',
      phone: '1234',
      profileImage: null,
      stripeCustomerId: null,
      invoiceAddress: null,
      createdAt: '2022-01-31T15:04:19.739Z',
      updatedAt: '2022-01-31T15:04:19.739Z',
    },
    invoice: null,
    receipt: null,
  },
  tipTransaction: null,
};
