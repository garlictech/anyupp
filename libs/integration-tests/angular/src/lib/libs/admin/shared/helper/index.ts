import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
