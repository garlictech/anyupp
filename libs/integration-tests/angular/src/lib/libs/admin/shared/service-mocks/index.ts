export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
