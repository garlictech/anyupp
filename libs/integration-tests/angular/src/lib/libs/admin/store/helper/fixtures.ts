export const testFixtures = [
  {
    id: 'test_fixture_id_1',
    name: 'Text fixture 1',
  },
  {
    id: 'test_fixture_id_2',
    name: 'Text fixture 2',
  },
  {
    id: 'test_fixture_id_3',
    name: 'Text fixture 3',
  },
];
