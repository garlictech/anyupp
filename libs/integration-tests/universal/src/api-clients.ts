import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import {
  getAnyuppSdkForIAM,
  getAnyuppSdkForUserPool,
} from '@bgap/anyupp-gql/api';
import { Auth } from 'aws-amplify';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';

Auth.configure({
  ...awsConfig,
  // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
  authenticationFlowType: 'USER_PASSWORD_AUTH',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
});

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));

export const createAuthenticatedAnyuppSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getAnyuppSdkForUserPool));

export const createIamCrudSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );

export const createIamAnyuppSdk = () =>
  getAnyuppSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
