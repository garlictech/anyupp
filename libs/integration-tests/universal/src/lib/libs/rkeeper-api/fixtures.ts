import * as CrudApi from '@bgap/crud-gql/api';
import {
  Dish,
  RKeeperBusinessEntityInfo,
  RkeeperFixtures,
} from '@bgap/rkeeper-api';
import { RequiredId } from '@bgap/shared/types';
import {
  unitFixture,
  productFixture,
  chainFixture,
  groupFixture,
  rkeeperEndpoint,
  yellowRestaurantId,
  yellowRkeeperUsername,
  yellowRkeeperPassword,
} from '@bgap/shared/fixtures';

const testIdPrefix = `rkeeper-cf0d1110-a2ce-45cf-aa69-6782bbc44cad`;

export const rawData = {
  data: {
    dishes: [
      RkeeperFixtures.dish,
      RkeeperFixtures.dish2,
      RkeeperFixtures.duplicatedDish,
      RkeeperFixtures.badDish,
      RkeeperFixtures.inactiveDish,
      RkeeperFixtures.dishWithModifier,
      RkeeperFixtures.dishWithSameModifier,
    ],
    modifiers: [RkeeperFixtures.modifier, RkeeperFixtures.unusedModifier],
  },
};

export const rkeeperProductGuid = 'RKEEPERGUID';

const unitId = `${testIdPrefix}-unit`;
const chainId = `${testIdPrefix}-chain`;
const groupId = `${testIdPrefix}-group`;

export const rkeeperUnit: RequiredId<CrudApi.CreateUnitInput> = {
  ...unitFixture.createRkeeperUnit,
  id: unitId,
  groupId,
  chainId,
  externalId: 'EXTERNAL-RESTAURANT-ID',
};

export const rkeeperUnitProduct: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct`,
  externalId: rkeeperProductGuid,
};

export const rkeeperUnitProduct2: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct2`,
  externalId: rkeeperProductGuid + '2',
};

export const processedDish: Dish = {
  price: 50000,
  active: true,
  id: 9991000114,
  guid: '999e3ab3-86a0-48d9-a9a9-f4e0c9fbce68',
  name: 'Pr√≥ba ital',
};

export const businessEntity: RKeeperBusinessEntityInfo = {
  chainId,
  groupId,
  unitId,
};

export const createChain: RequiredId<CrudApi.CreateChainInput> = {
  ...chainFixture.chainBase,
  id: chainId,
};

export const createGroup: RequiredId<CrudApi.CreateGroupInput> = {
  ...groupFixture.groupBase,
  id: groupId,
  chainId,
};

export const rkeeperOrder = {
  objectid: yellowRestaurantId,
  remoteId: yellowRestaurantId,
  order_type: 1,
  pay_type: 0,
  pay_online_type: 0,
  delivery_time: '2122-02-03T12:37:32',
  client: {
    phone: null,
    email: null,
    ln: 'Testln',
    fn: 'Testfn',
  },
  order: [
    {
      id: '1000114',
      type: 'd',
      qnt: 1000,
    },
  ],
};

export const yellowUnit = {
  ...rkeeperUnit,
  externalId: yellowRestaurantId,
  pos: {
    ...rkeeperUnit.pos,
    rkeeper: {
      ...(rkeeperUnit?.pos?.rkeeper || {}),
      rkeeperUsername: yellowRkeeperUsername,
      rkeeperPassword: yellowRkeeperPassword,
      endpointUri: rkeeperEndpoint,
    } as CrudApi.RKeeper,
  },
} as CrudApi.Unit;

export const orderInput: CrudApi.CreateOrderInput = {
  userId: 'USER_ID',
  unitId: yellowUnit.id,
  items: [
    {
      quantity: 5,
      productId: 'PRODUCT ID',
      statusLog: [
        {
          userId: 'USER_ID',
          status: CrudApi.OrderStatus.none,
          ts: 1627909024677,
        },
      ],
      sumPriceShown: {
        taxSum: 316.98,
        currency: 'HUF',
        tax: 27,
        priceSum: 1491,
        pricePerUnit: 298.2,
      },
      productName: {
        hu: 'PRODUCT NAME HU',
      },
      priceShown: {
        taxSum: 318.9,
        currency: 'HUF',
        tax: 27,
        priceSum: 1500,
        pricePerUnit: 300,
      },
      variantId: 'VARIANT ID',
      variantName: {
        en: 'VARIANT NAME EN',
      },
    },
  ],
  sumPriceShown: {
    taxSum: 633.96,
    currency: 'HUF',
    tax: 27,
    priceSum: 2982,
    pricePerUnit: 298.2,
  },
  place: {
    table: '01',
    seat: '01',
  },
  orderMode: CrudApi.OrderMode.instant,
  servingMode: CrudApi.ServingMode.inplace,
  archived: false,
  paymentMode: {
    method: CrudApi.PaymentMethod.cash,
    type: CrudApi.PaymentType.cash,
  },
};

export {
  yellowRestaurantId,
  yellowRkeeperPassword,
  yellowRkeeperUsername,
  rkeeperEndpoint,
};
