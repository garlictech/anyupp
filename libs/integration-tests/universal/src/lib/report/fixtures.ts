import { DateTime } from 'luxon';

import { timezoneBudapest } from '@bgap/shared/utils';

const dateAgo = (days: number) =>
  DateTime.fromISO(new Date().toISOString(), {
    zone: timezoneBudapest,
  })
    .minus({ days })
    .toJSDate()
    .toString();

export const reportUsersFixture = [
  {
    id: 'anonym_1',
    email: 'testuser+anonymuser1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_2',
    email: 'testuser+anonymuser2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_3',
    email: 'testuser+anonymuser3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_1',
    email: 'testuser+a1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_2',
    email: 'testuser+a2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_3',
    email: 'testuser+a3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_1',
    email: 'testuser+fb1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_2',
    email: 'testuser+fb2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_1',
    email: 'testuser+g1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_2',
    email: 'testuser+g2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_1',
    email: 'testuser+apple1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_2',
    email: 'testuser+apple2@anyupp.com',
    ucd: new Date().getTime(),
  },
];

export const fourWeeksOrdersFixture = [
  {
    createdAt: dateAgo(1),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(2),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(3),
    userId: 'anonym_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(4),
    userId: 'anonym_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(5),
    userId: 'email_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(6),
    userId: 'email_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(7),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(8),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(9),
    userId: 'facebook_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(10),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];

export const thisYearOrdersFixture = [
  ...fourWeeksOrdersFixture,
  {
    createdAt: dateAgo(11),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(42),
    userId: 'google_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(43),
    userId: 'google_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(44),
    userId: 'signinwithapple_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(45),
    userId: 'signinwithapple_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];
