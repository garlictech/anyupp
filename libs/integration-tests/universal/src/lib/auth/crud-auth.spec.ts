import * as CrudApi from '@bgap/crud-gql/api';
import { orderFixture } from '@bgap/shared/fixtures';
import { createIamCrudSdk } from '../../api-clients';

describe('CRUD endpoints AUTH test', () => {
  const crudSdk = CrudApi.getCrudSdkPublic();
  const iamSdk = createIamCrudSdk();

  it('should require authentication to access', done => {
    crudSdk.GetOrder({ id: orderFixture.order_seeded_01_id }).subscribe({
      error(e) {
        expect(e).toMatchSnapshot();
        done();
      },
    });
  }, 15000);

  describe('IAM Auth', () => {
    it('should be able to execute a query with IAM authenticated graphql client', done => {
      iamSdk.GetOrder({ id: orderFixture.order_seeded_01_id }).subscribe({
        next(x) {
          expect(x?.id).toEqual(orderFixture.order_seeded_01_id);
          done();
        },
      });
    }, 15000);
  });
});
