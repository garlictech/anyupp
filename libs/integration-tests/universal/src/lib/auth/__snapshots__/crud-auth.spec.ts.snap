// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUD endpoints AUTH test should require authentication to access 1`] = `[Error: GraphQL error: Not Authorized to access getOrder on type Query]`;
